### PR TITLE
Feature/player damage

### PR DESCRIPTION
**Pull request by lssalgado**
* Organizing Mob scene structure into folders;
* Added Player lives, ability to taka damage and die.

### CODE REVIEW

File: scripts/Entity.gd [hunk 1]
```diff
@@ -26,6 +28,9 @@ func jump():
 	motion.y = -jump_force
 	resistance = AIR_RESISTANCE
 
+func take_damage():
```
Comment (lssalgado) [inline scripts/Entity.gd]: There should be a common code for taking damage, like `lives -= 1`.

Also, this method should receive an `int` variable as a parameter saying the damage value.

File: scripts/player/player.gd [hunk 2]
```diff
@@ -2,15 +2,31 @@ extends Entity
 
 class_name Player
 
+signal player_dead
```
Comment (lssalgado) [inline scripts/player/player.gd:5]: I think we should follow Godot naming pattern for signals and name it something like `on_player_death`.

File: scripts/player/player.gd [hunk 3]
```diff
@@ -2,15 +2,31 @@ extends Entity
...
 	print("Damage Taken")
+	
+	lives -= 1
+	
+	if lives <= 0:
+		emit_signal("player_dead")
+	else:
+		animation_effects.play("damage")
+		animation_effects.queue("flash")
+		damage_timer.start()
```
Comment (lssalgado) [inline scripts/player/player.gd]: This timer is way too long. I think we could lower it to 1 second, at maximum.

File: scripts/player/player.gd [hunk 4]
```diff
@@ -2,15 +2,31 @@ extends Entity
...
+		invulnerable = true
+		set_collision_mask_bit(2, false)
```
Comment (lssalgado) [inline scripts/player/player.gd]: What does 2 mean?

File: scripts/player/player.gd (hunk 2 above)
Comment (uSirThunder) [reply scripts/player/player.gd:5]: player_dead is not a method, just a signal.
The method for the signal would be called on_player_dead.

File: scripts/player/player.gd (hunk 3 above)
Comment (uSirThunder) [reply scripts/player/player.gd]: Timer ajusted to 0.8 sec

File: scripts/player/player.gd (hunk 4 above)
Comment (uSirThunder) [reply scripts/player/player.gd]: 2 is the ID to the collision mask bit.

**Pull request merged**